org.eclipse.scanning.sequencer/src/org/eclipse/scanning/sequencer/AbstractRunnableDeviceDelegate.java

```diff
@@ -0,0 +1,183 @@
+		return new NexusObjectWrapper<NXdetector>(name, nexusBaseClass);
+	}
+
+	// Delegated annotated methods
+
+	@SuppressWarnings("unused")
+	public void preConfigure(Object scanModel, ScanBean scanBean, IPublisher<?> publisher) throws ScanningException {
+		logger.trace("preConfigure({}, {}, {}) on {}", scanModel, scanBean, publisher, runnableDeviceProxy.getName());
+	}
+
+	@SuppressWarnings("unused")
+	public void postConfigure(Object scanModel, ScanBean scanBean, IPublisher<?> publisher) throws ScanningException {
+		logger.trace("postConfigure({}, {}, {}) on {}", scanModel, scanBean, publisher, runnableDeviceProxy.getName());
+	}
+
+	@SuppressWarnings("unused")
+	public void levelStart(LevelInformation info)  throws ScanningException { // Other arguments are allowed
+		logger.trace("levelStart({}) on {}", info, runnableDeviceProxy.getName());
+	}
+
+	@SuppressWarnings("unused")
+	public void levelEnd(LevelInformation info) throws ScanningException { // Other arguments are allowed
+		logger.trace("levelEnd({}) on {}", info, runnableDeviceProxy.getName());
+	}
+
+	@SuppressWarnings("unused")
+	public void pointStart(IPosition point) throws ScanningException {
+		logger.trace("pointStart({}) on {} stepIndex={}", point, runnableDeviceProxy.getName(), point.getStepIndex());
+	}
+
+	@SuppressWarnings("unused")
+	public void pointEnd(IPosition point) throws ScanningException {
+		logger.trace("pointEnd({}) on {} stepIndex={}", point, runnableDeviceProxy.getName(), point.getStepIndex());
+	}
+
+	@SuppressWarnings("unused")
+	public void scanStart(ScanInformation info) throws ScanningException {
+		logger.trace("scanStart({}) on {} filePath={}", info, runnableDeviceProxy.getName(), info.getFilePath());
+	}
+
+	@SuppressWarnings("unused")
+	public void scanEnd(ScanInformation info) throws ScanningException {
+		logger.trace("scanEnd({}) on {} filePath={}", info, runnableDeviceProxy.getName(), info.getFilePath());
+	}
+
+	/** Delegated {@link WriteComplete}
+	 *
+	 * @param info
+	 * @throws ScanningException
+	 */
+	@SuppressWarnings("unused")
+	public void writeComplete(ScanInformation info) throws ScanningException {
+		logger.trace("writeComplete({}) on {} filePath={}", info, runnableDeviceProxy.getName(), info.getFilePath());
+	}
+
+	@SuppressWarnings("unused")
+	public void scanAbort(ScanInformation info) throws ScanningException {
+		logger.trace("scanAbort({}) on {}", info, runnableDeviceProxy.getName());
+	}
+
+	@SuppressWarnings("unused")
+	public void scanFault(ScanInformation info) throws ScanningException {
+		logger.trace("scanFault({}) on {}", info, runnableDeviceProxy.getName());
+	}
+
+	@SuppressWarnings("unused")
+	public void scanFinally(ScanInformation info) throws ScanningException {
+		logger.trace("scanFinally({}) on {}", info, runnableDeviceProxy.getName());
+	}
+
+	@SuppressWarnings("unused")
+	public void scanPause() throws ScanningException {
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Remove the declaration of thrown exception 'org.eclipse.scanning.api.scan.ScanningException', as it cannot be thrown from method's body. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3ARedundantThrowsDeclarationCheck)